### ISSUE TITLE
Daemon, Worker process hang on Ubuntu

### ISSUE THREAD

**Issue by vvisigoth** [urbit/urbit#1702]
Situations arise during %hole/%crud event replacement that causes the process to hang and become unresponsive. killing the process and restarting will sometimes work for a period of time, but will eventually try to replace an event again and hang. Attached gdb to the processes reveals that they are simply waiting for one another. 

This seems to be particularly exacerbated post-individual breach, due to the number of failed events coming out of gall/ames (speculation).

@joemfb and I discovered yesterday that this appears to only happen on Linux (Ubuntu). I tarred a hanging pier and am currently running it on an OSX laptop with no problems.

@joemfb and @benjamin-tlon seem to have an inkling of the issue here (something something file descriptors) and are working on it. I am recording here for posterity.

TL;DR moving pier to a different platform may allow you to get around the hang temporarily, but we're working on a real fix.

**Comment by vvisigoth** [comment]
@joemfb @benjamin-tlon I was able to reproduce this on ~poldec on the same GCP. @benjamin-tlon can you try the fix we discussed and we'll try to see if it fixes.

**Comment by vvisigoth** [comment]
By "stuck" do you mean "unresponsive to keyboard input"? Then yes.
—
~poldec-tonteg
http://urbit.org

**Comment by joemfb** [comment]
@vvisigoth, I don't think this has been reproduced. Good to close?

**Comment by vvisigoth** [comment]
Yes
—
~poldec-tonteg
http://urbit.org
